Common function ( in all teams, out teams with available places)

stevenTCL opened the pull request:
- function (input: all teams, output: filtered teams that still have places left)
- unit test passed

stevenTCL committed:
getAvailableTeam_Function

- done getAvailableTeam (teams)
- done unit test file
- NOT YET TESTED

stevenTCL committed:
UserTestPassed

1. bugs fixed
2. user test passed

stevenTCL committed:
typo

stevenTCL committed:
update coverage

chautsunman commented on app/js/common.js:
```diff
@@ -0,0 +1,8 @@
+var getAvailableTeam = function (teams) {
+	var output = [];
+	teams.forEach(function(team) {
```
use Array.filter() (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

stevenTCL committed:
filter done

1. remove previous package.json
2. refactor the function

chautsunman commented on app/js/common.js:
```diff
@@ -0,0 +1,7 @@
+	return team.currentTeamSize < team.size;
+}
+
+var getAvailableTeam = function (teams) {
```
add documentation comment

```
/**
 * filter teams that still have places left
 *
 * @param teams array of teams
 * @return filtered teams
 */
```

stevenTCL committed:
syntax stuff

1. deleted the new line added in .json, there's still a new line of
firefox-launcher
2. used standard function syntax  `function functionName(parameters) {}`
3. added the documentation comment
